software/control/microcontroller.py

```diff
@@ -630,21 +639,20 @@ def read_received_packet(self):
             self._cmd_id_mcu = msg[0]
             self._cmd_execution_status = msg[1]
             if (self._cmd_id_mcu == self._cmd_id) and (self._cmd_execution_status == CMD_EXECUTION_STATUS.COMPLETED_WITHOUT_ERRORS):
-                if self.mcu_cmd_execution_in_progress == True:
+                if self.mcu_cmd_execution_in_progress:
                     self.mcu_cmd_execution_in_progress = False
                     self.log.debug("mcu command " + str(self._cmd_id) + " complete")
-            elif self._cmd_id_mcu != self._cmd_id and time.time() - self.last_command_timestamp > 5 and self.last_command != None:
-                self.timeout_counter = self.timeout_counter + 1
-                if self.timeout_counter > 10:
+            elif self.mcu_cmd_execution_in_progress and self._cmd_id_mcu != self._cmd_id and time.time() - self.last_command_send_timestamp > self.LAST_COMMAND_ACK_TIMEOUT and self.last_command is not None:
+                if self.retry > self.MAX_RETRY_COUNT:
+                    self.abort_current_command(reason=f"Command timed out without an ack after {self.LAST_COMMAND_ACK_TIMEOUT} [s], and {self.retry} retries")
```

Comment:
After this, callers need to acknowledge the aborted command to not get warnings.  We won't stop them from sending commands without ack-ing, for now, though.